src/libs/metadata_view.c

```diff
@@ -791,6 +807,115 @@ void connect_key_accels(dt_lib_module_t *self)
   dt_accel_connect_lib(self, "jump to film roll", closure);
 }
 
+static void _update_layout(dt_lib_module_t *self)
+{
+  dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)self->data;
+  char *pref = dt_conf_get_string("plugins/lighttable/metadata_view/pref_hidden");
+  GList *prefs = dt_util_str_to_glist(",", pref);
+  g_free(pref);
+  for(int i = 0; i < md_size; i++)
+  {
+    const gboolean hidden = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0) != NULL;
+    for(int j = 0; j < 2; j++)
+    {
+      GtkWidget *w = gtk_grid_get_child_at(GTK_GRID(d->grid), j, i);
+      gtk_widget_show_all(w);
+      gtk_widget_set_no_show_all(w, TRUE);
+      gtk_widget_set_visible(w, !hidden);
+    }
+  }
+  g_list_free_full(prefs, g_free);
+}
+
+void set_preferences(dt_lib_module_t *self)
+{
+  dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)self->data;
+
+  GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("metadata settings"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
+                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
+
+  GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
+  gtk_widget_set_size_request(w, -1, DT_PIXEL_APPLY_DPI(300));
+  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+  gtk_container_add(GTK_CONTAINER(area), w);
+
+  GtkWidget *grid = gtk_grid_new();
+  gtk_container_add(GTK_CONTAINER(w), grid);
+  GtkWidget *label = gtk_label_new(_("metadata"));
+  gtk_grid_attach(GTK_GRID(grid), label, 0, 0, 1, 1);
+  label = gtk_label_new(_("hidden"));
+  gtk_grid_attach(GTK_GRID(grid), label, 1, 0, 1, 1);
+  gtk_widget_set_tooltip_text(GTK_WIDGET(label),
+            _("tick if the corresponding metadata is of no interest for you"));
+
+  char *pref = dt_conf_get_string("plugins/lighttable/metadata_view/pref_hidden");
+  GList *prefs = dt_util_str_to_glist(",", pref);
+  g_free(pref);
+  GtkWidget *hidden[md_size];
+  gboolean hide[md_size];
+  for(int i = 0; i < md_size; i++)
+  {
+    label = gtk_label_new(_md_labels[i]);
```

Comment:
could you use dt_ui_label_new? it is missing `gtk_label_set_xalign (label, 0.0f);`  which should be added there.

Reply:
Yes for the first. Already there for the second.
Removed what is already in dt_ui_label_new.

Reply:
I meant you could probably add gtk_label_set_xalign to dt_ui_label_new, because generally halign and xalign go hand in hand. But you can also leave it here.

Reply:
> you could probably add gtk_label_set_xalign to dt_ui_label_new

done. I hope that won't create issue elsewhere...